source/wp-content/plugins/wporg-post-title/wporg-post-title.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Plugin Name:       Wporg Post Title Block
+ * Description:       Post title block with custom attributes needed by WordPress.org.
+ * Author:            WordPress.org
+ * Author URI:        https://wordpress.org/
+ * License:           GPL v2 or later
+ * License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+ * Text Domain:       wporg
+ */
+
+/**
+ * Server-side rendering of the `wporg/post-title` block.
+ */
+
+/**
+ * Renders the `wporg/post-title` block on the server.
+ * Copied from core post-title.php and modified.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the filtered post title for the current post wrapped inside "h1" tags.
+ */
+function render_block_wporg_post_title( $attributes, $content, $block ) {
```

Comment:
I was imagining something much more simple, that would only be used for this purpose, since `core/post-title` can be used if they want the full title.

```php
register_block_type(
	'wporg/post-title-version',
	array(
		'title'           => 'Post Title Version',
		'render_callback' => function() {
			$full_title = get_the_title();
			preg_match( '/WordPress (\d+[.]\d+)/', $full_title, $matches );
			
			return $matches[1] ?? ''; 
		}
	)
);
```

That's probably missing a few things, but communicates the jist of it. I don't think we need `block.json` etc for a small workaround like this. In my mind this is just a temporary thing until  Gutenberg solves https://github.com/WordPress/gutenberg/issues/32939.